src/ValueFormatters/Exceptions/MismatchingDataValueTypeException.php

```diff
@@ -29,14 +30,19 @@ class MismatchingDataValueTypeException extends FormattingException {
 	 * @param string $message
 	 * @param Exception|null $previous
 	 */
-	public function __construct( $expectedValueType, $dataValueType, $message = '',
+	public function __construct(
+		$expectedValueType,
+		$dataValueType,
+		$message = '',
 		Exception $previous = null
 	) {
 		$this->expectedValueType = $expectedValueType;
 		$this->dataValueType = $dataValueType;
 
-		$message = '$dataValueType "' . $dataValueType . '" does not match $expectedValueType "'
-			. $expectedValueType . '": $message';
+		if ( $message === '' ) {
```

Comment:
Because this feature is unused anyway and also provides no benefit. Other exceptions do have similar code. They all (as far as I'm aware of) do the same: Use the custom message, if provided. Construct a message otherwise. But do not combine the two.